### PR TITLE
Construct the accessibility channel's events by SemanticsEvent.

### PR DESCRIPTION
**Pull request by amirh**
This refactoring allows us to have SemanticsEvent object for events that are not
associated with an accessibility node id.
And allow https://github.com/flutter/flutter/pull/12594 to be a bit
cleaner with a single place for accessibility channel documentation (the
SemanticsEvent classes documentation).

### CODE REVIEW

File: packages/flutter/lib/src/rendering/semantics_event.dart [hunk 1]
```diff
@@ -25,15 +25,27 @@ abstract class SemanticsEvent {
...
   /// [StandardMessageCodec].
-  Map<String, dynamic> toMap();
+  Map<String, dynamic> toMap({int nodeId}) {
+    final Map<String, dynamic> event = <String, dynamic>{
+      'type': type,
+      'data': getDataMap(),
+    };
+    if (nodeId != null) {
+      event['nodeId'] = nodeId;
+    }
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/semantics_event.dart]: nit: no { } for one-liner

File: packages/flutter/lib/src/rendering/semantics_event.dart [hunk 2]
```diff
@@ -25,15 +25,27 @@ abstract class SemanticsEvent {
 
   /// Converts this event to a Map that can be encoded with
   /// [StandardMessageCodec].
-  Map<String, dynamic> toMap();
+  Map<String, dynamic> toMap({int nodeId}) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/semantics_event.dart]: nit: spaces after { and before }

File: packages/flutter/lib/src/rendering/semantics_event.dart [hunk 3]
```diff
@@ -25,15 +25,27 @@ abstract class SemanticsEvent {
...
+      'type': type,
+      'data': getDataMap(),
+    };
+    if (nodeId != null) {
+      event['nodeId'] = nodeId;
+    }
+    return event;
+  }
+
+  // Returns the event's data object.
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/semantics_event.dart]: Three leading ///.

Should also mention that the returned obj needs to be encodable with [StandardMessageCodec].

File: packages/flutter/lib/src/rendering/semantics_event.dart (hunk 2 above)
Comment (goderbauer) [reply packages/flutter/lib/src/rendering/semantics_event.dart]: This should also document when and what the nodeId is.

**Comment by amirh** [comment]
PTAL

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**